py_swf/clients/workflow.py

```diff
@@ -65,3 +65,344 @@ def terminate_workflow(self, workflow_id, reason):
+        """
+        close_status_filter_dict = _build_close_status_filter_dict(status)
+        return self._count_closed_workflow_executions(
+            oldest_close_date=oldest_close_date,
+            latest_close_date=latest_close_date,
+            workflow_filter_dict=close_status_filter_dict,
+        )
+
+    def _count_open_workflow_executions(
+            self,
+            oldest_start_date=None,
+            latest_start_date=None,
+            workflow_filter_dict=None
+    ):
+        """
+        Count the number of open workflows for a domain. You can pass in filter criteria as a dict
+        The results are best effort and may not exactly reflect recent updates and changes.
+        http://docs.aws.amazon.com/amazonswf/latest/apireference/API_CountOpenWorkflowExecutions.html
+        :param oldest_start_date: datetime
+        :param latest_start_date: datetime
+        :param workflow_filter_dict: dict. filter type as key and filter criteria as value. filter criteria is a dictionary
+        :return: Returns the number of open workflow executions within the given domain that meet the specified filtering criteria.
+        """
+        if not workflow_filter_dict:
+            workflow_filter_dict = {}
+
+        start_time_filter_dict = _build_time_filter_dict(
+            oldest_start_date=oldest_start_date,
+            latest_start_date=latest_start_date,
+        )
+        if not start_time_filter_dict:
+            return self.boto_client.count_open_workflow_executions(
+                domain=self.workflow_client_config.domain,
+                **workflow_filter_dict
+            )
+
+        return self.boto_client.count_open_workflow_executions(
+            domain=self.workflow_client_config.domain,
+            startTimeFilter=start_time_filter_dict['startTimeFilter'],
+            **workflow_filter_dict
+        )
+
+    def _count_closed_workflow_executions(
+            self,
+            oldest_start_date=None,
+            latest_start_date=None,
+            oldest_close_date=None,
+            latest_close_date=None,
+            workflow_filter_dict=None
+    ):
+        """
+        Count the number of closed workflows for a domain. You can pass in filtering criteria as a dict
+        The results are best effort and may not exactly reflect recent updates and changes.
+        http://docs.aws.amazon.com/amazonswf/latest/apireference/API_CountClosedWorkflowExecutions.html#SWF-CountClosedWorkflowExecutions-request-executionFilter
+        :param oldest_start_date: datetime
+        :param latest_start_date: datetime
+        :param oldest_close_date: datetime
+        :param latest_close_date: datetime
+        :param workflow_filter_dict: dict. filter type as key and filter criteria as value. filter criteria is a dictionary
+        :return: total number of closed workflows that meet the filter criteria
+        """
+        if not workflow_filter_dict:
```

Comment:
nit: explicitly check for `is None`. Also, better to inline these as kwargs, instead of having to instantiate a dictionary that will be passed in as kwargs anyway to the underlying count function.